test_port_allocation.sh

```
@@ -0,0 +1,75 @@
+
+# Source the run_bot.sh to get access to the port functions
+source ./run_bot.sh > /dev/null 2>&1
+
+# Test 1: Sequential port allocation
+echo "Test 1: Sequential port allocation"
+echo "-----------------------------------"
+
+for i in {1..5}; do
+    echo "Instance $i:"
+    ports=$(get_next_bot_ports)
+    if [ $? -eq 0 ]; then
```

Shellcheck (0.11.0)

[style] Check exit code directly with e.g. 'if mycmd;', not indirectly with $?.

(SC2181)